cpp/include/raft/neighbors/detail/cagra/cagra_build.cuh

```diff
@@ -296,7 +296,8 @@ index<T, IdxT> build(
   std::optional<experimental::nn_descent::index_params> nn_descent_params = std::nullopt,
   std::optional<float> refine_rate                                        = std::nullopt,
   std::optional<ivf_pq::index_params> pq_build_params                     = std::nullopt,
-  std::optional<ivf_pq::search_params> search_params                      = std::nullopt)
+  std::optional<ivf_pq::search_params> search_params                      = std::nullopt,
+  bool construct_index_with_dataset                                       = true)
```

Comment:
Currently only `cagra::detail::build` is changed. This is enough to unblock benchmarking with RAFT-ANN-Bench, but ideally we should have such an option in the public API. Opinions?